src/os/bluestore/BlueFS.cc

```diff
@@ -1868,6 +1868,8 @@ int BlueFS::_read(
     if (off < buf->bl_off || off >= buf->get_buf_end()) {
       s_lock.unlock();
       std::unique_lock u_lock(h->lock);
+      //We will scan through fnode.extents, and use fnode.size. Must protect against writers.
+      std::shared_lock fnode_lock(h->file->lock);
```

Comment:
shouldn't we take the same lock in _read_random as it calls h->file->fnode.seek as well?

Reply:
True.